src/utp_stream.cpp

```diff
@@ -210,6 +210,152 @@ struct packet
 	std::uint8_t buf[1];
 };
 
+// single thread packet allocation packet pool
+// can handle 3 common cases of packet allocation size
+struct packet_pool
```

Comment:
this class needs a unit test, which I believe requires it to be defined in a header file. ideally it's own header file